lib/background_worker/base.rb

```diff
@@ -72,109 +65,58 @@ def perform_in_background(method_name, options={})
         # Enqueue to the background queue
         BackgroundWorker.enqueue(self, method_name, options)
 
-        return options[:uid]
+        options[:uid]
       end
 
-
-      def rescue_reconnect
-        Resque.redis.client.disconnect
-        Resque.redis.client.reconnect
-        Rails.cache.reconnect
-      end
-
-
       # This method is called by the job runner
       #
       # It will just call your preferred method in the worker.
-      def perform(method_name, options={})
-        rescue_reconnect if defined?(Resque)
-        ActiveRecord::Base.verify_active_connections!
+      def perform(method_name, options = {})
+        fail ArgumentError, ":uid is required: Options given were: #{options.inspect}" if options['uid'].blank?
+        BackgroundWorker.verify_active_connections!
 
-        raise ArgumentError, ":uid is required: Options given were: #{options.inspect}" if options['uid'].blank?
-        setup_logger(method_name)
-        raise ArgumentError, ":current_user_id is required: Options given were: #{options.inspect}" if options['current_user_id'].blank?
+        # Special 'user' handling
+        fail ArgumentError, ":current_user_id is required: Options given were: #{options.inspect}" if options['current_user_id'].blank?
```

Comment:
Line is too long. [135/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.